airbyte-ci/connectors/pipelines/pipelines/tests/java_connectors.py

```diff
@@ -107,9 +103,15 @@ async def run_all_tests(context: ConnectorContext) -> List[StepResult]:
     connector_container = build_connector_image_results.output_artifact[LOCAL_BUILD_PLATFORM]
     connector_image_tar_file, _ = await export_container_to_tarball(context, connector_container)
 
-    integration_tests_results = await IntegrationTests(context).run(connector_image_tar_file, normalization_tar_file)
-    step_results.append(integration_tests_results)
+    async with asyncer.create_task_group() as test_task_group:
+        soon_unit_tests_results = test_task_group.soonify(UnitTests(context).run)()
```

Comment:
These don't depend on building and loading the connector image. You might get even better performance (due to gradle caching) if you kicked off this task as soon as the distTar is built. Possibly sooner. 

Perhaps this is out of scope for this work but it would be nice to take a more concurrent approach here: kick off each of these steps ASAP and have each of them wait on their dependencies being fulfilled.

Reply:
Thanks for this suggestion, trying this out right now.

Reply:
[It speeds things up by 1mn!](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/manual/augustin_10-15-airbyte-ci_concurrent_java_tests/1697486713/e107b1377760da6036a7e3697e043b83f53c5339/source-postgres/3.1.13/output.html)